lib/openstax/aws/stack.rb

```diff
@@ -325,16 +378,20 @@ def self.format_hash_as_tag_parameters(tags)
       tags.map{|tag| {key: tag.key, value: tag.value}}
     end
 
-    def status
-      begin
-        aws_stack.stack_status
-      rescue Aws::CloudFormation::Errors::ValidationError => ee
-        case ee.message
-        when /Stack.*does not exist/
-          "DOES_NOT_EXIST"
-        else
-          raise
-        end
+    def status(reload: false)
+      @status = nil if reload
+      @status ||= Status.new(self)
+    end
+
+    def events
+      aws_stack&.events || []
```

Comment:
Since we have an `Event` class of our own now, let's return objects of that type here

```ruby
  (aws_stack&.events || []).each {|aws_event| Event.new(aws_event)}
```